runtime/jvmti/jvmtiWatchedField.c

```diff
@@ -126,44 +131,59 @@ setFieldWatch(jvmtiEnv* env,
 
 		vm->internalVMFunctions->acquireExclusiveVMAccess(currentThread);
 
-		/* Scan currently watched fields to see if this agent is already watching */
-
-		watchedField = pool_startDo(j9env->watchedFieldPool, &poolState);
-		while (watchedField != NULL) {
-			if (watchedField->fieldID == field) {
-				break;
+		clazz = J9VM_J9CLASS_FROM_JCLASS(currentThread, klass);
+		fieldID = (J9JNIFieldID*)field;
+		localFieldIndex = fieldID->index - fieldID->declaringClass->romClass->romMethodCount;
+		fieldCount = clazz->romClass->romFieldCount;
+		watchedClass = hashTableFind(j9env->watchedClasses, &clazz);
+		if (NULL == watchedClass) {
+			J9JVMTIWatchedClass exemplar = { clazz, NULL };
+			watchedClass = hashTableAdd(j9env->watchedClasses, &exemplar);
+			if (NULL == watchedClass) {
+				rc = JVMTI_ERROR_OUT_OF_MEMORY;
+			} else {
+				UDATA fieldCount = clazz->romClass->romFieldCount;
+				if (fieldCount <= J9JVMTI_WATCHED_FIELDS_PER_UDATA) {
+					watchedClass->watchBits = (UDATA*)0;
+				} else {
+					UDATA allocSize = sizeof(UDATA) * J9JVMTI_WATCHED_FIELD_ARRAY_INDEX(fieldCount + (J9JVMTI_WATCHED_FIELDS_PER_UDATA - 1));
+					PORT_ACCESS_FROM_VMC(currentThread);
+					watchBits = j9mem_allocate_memory(allocSize, J9MEM_CATEGORY_JVMTI);
+					if (NULL == watchBits) {
+						hashTableRemove(j9env->watchedClasses, watchedClass);
+						rc = JVMTI_ERROR_OUT_OF_MEMORY
```

Comment:
you appear to be missing a semi-colon on this line.

Reply:
Compiler agrees with you - fixed.